src/common/meta/src/key/topic_region.rs

```diff
@@ -0,0 +1,228 @@
+                ),
+            }
+            .build()
+        })?;
+        TopicRegionKey::try_from(key)
+    }
+}
+
+impl Display for TopicRegionKey<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}/{}/{}",
+            TOPIC_REGION_PREFIX,
+            self.topic,
+            self.region_id.as_u64()
+        )
+    }
+}
+
+impl<'a> TryFrom<&'a str> for TopicRegionKey<'a> {
+    type Error = Error;
+
+    /// Value is of format `{prefix}/{topic}/{region_id}`
+    fn try_from(value: &'a str) -> Result<TopicRegionKey<'a>> {
+        let captures = TOPIC_REGION_PATTERN
+            .captures(value)
+            .context(InvalidMetadataSnafu {
+                err_msg: format!("Invalid TopicRegionKey: {}", value),
+            })?;
+        let topic = captures.get(1).map(|m| m.as_str()).unwrap();
+        let region_id = captures[2].parse::<u64>().map_err(|_| {
+            InvalidMetadataSnafu {
+                err_msg: format!("Invalid region id in TopicRegionKey: {}", value),
+            }
+            .build()
+        })?;
+        Ok(TopicRegionKey {
+            region_id: RegionId::from_u64(region_id),
+            topic,
+        })
+    }
+}
+
+fn topic_region_decoder(value: &KeyValue) -> Result<TopicRegionKey<'_>> {
+    let key = TopicRegionKey::from_bytes(&value.key)?;
+    Ok(key)
+}
+
+/// Manages map of topics and regions in kvbackend.
+pub struct TopicRegionManager {
+    kv_backend: KvBackendRef,
+}
+
+impl TopicRegionManager {
+    pub fn new(kv_backend: KvBackendRef) -> Self {
+        Self { kv_backend }
+    }
+
+    pub async fn put(&self, key: TopicRegionKey<'_>) -> Result<()> {
+        let put_req = PutRequest {
+            key: key.to_bytes(),
+            value: vec![],
+            prev_kv: false,
+        };
+        self.kv_backend.put(put_req).await?;
+        Ok(())
+    }
+
+    pub async fn batch_put(&self, keys: Vec<TopicRegionKey<'_>>) -> Result<()> {
+        let req = BatchPutRequest {
+            kvs: keys
+                .into_iter()
+                .map(|key| KeyValue {
+                    key: key.to_bytes(),
+                    value: vec![],
+                })
+                .collect(),
+            prev_kv: false,
+        };
+        self.kv_backend.batch_put(req).await?;
+        Ok(())
+    }
+
+    /// Returns the list of region ids using specified topic.
+    pub async fn regions(&self, topic: &str) -> Result<Vec<RegionId>> {
+        let prefix = TopicRegionKey::range_topic_key(topic);
+        let req = RangeRequest::new().with_prefix(prefix.as_bytes());
+        let resp = self.kv_backend.range(req).await?;
+        let region_ids = resp
+            .kvs
+            .iter()
+            .map(topic_region_decoder)
+            .collect::<Result<Vec<_>>>()?;
+        Ok(region_ids.iter().map(|key| key.region_id).collect())
+    }
+
+    pub async fn delete(&self, key: TopicRegionKey<'_>) -> Result<()> {
```

Comment:
Do we also need a `delete range` by topic prefix?

Reply:
I'm not sure if there're cases we need to delete a topic in remote wal. We can add it if it's needed in the future.